modin/core/execution/ray/common/deferred_execution.py

```diff
@@ -0,0 +1,786 @@
+    func : callable or ObjectRefType
+        A function to be executed.
+    args : list or tuple
+        Additional positional arguments to be passed in `func`.
+    kwargs : dict
+        Additional keyword arguments to be passed in `func`.
+    num_returns : int, optional
+        The number of the return values.
+
+    Attributes
+    ----------
+    data : ObjectRefType or DeferredExecution
+        The execution input.
+    func : callable or ObjectRefType
+        A function to be executed.
+    args : list or tuple
+        Additional positional arguments to be passed in `func`.
+    kwargs : dict
+        Additional keyword arguments to be passed in `func`.
+    num_returns : int
+        The number of the return values.
+    flat_args : bool
+        True means that there are no lists or DeferredExecution objects in `args`.
+        In this case, no arguments processing is performed and `args` is passed
+        to the remote method as is.
+    flat_kwargs : bool
+        The same as `flat_args` but for the `kwargs` values.
+    """
+
+    def __init__(
+        self,
+        data: Union[
+            ObjectRefType,
+            "DeferredExecution",
+            List[Union[ObjectRefType, "DeferredExecution"]],
+        ],
+        func: Union[Callable, ObjectRefType],
+        args: Union[List[Any], Tuple[Any]],
+        kwargs: Dict[str, Any],
+        num_returns=1,
+    ):
+        self.data = data
+        self.func = func
+        self.args = args
+        self.kwargs = kwargs
+        self.num_returns = num_returns
+        de_or_list = DeOrListOrTuple
+        self.flat_args = not any(isinstance(a, de_or_list) for a in args)
+        self.flat_kwargs = not any(isinstance(a, de_or_list) for a in kwargs.values())
+        self.subscribers = 0
+
+    def exec(
+        self,
+    ) -> Tuple[ObjectRefOrListType, Union["MetaList", List], Union[int, List[int]]]:
+        """
+        Execute this task, if required.
+
+        Returns
+        -------
+        tuple
+            The execution result, MetaList, containing the length, width and
+            the worker's ip address (the last value in the list) and the values
+            offset in the list. I.e. length = meta_list[offset],
+            width = meta_list[offset + 1], ip = meta_list[-1].
+        """
+        if self.has_result:
+            return self.data, self.meta, self.meta_offset
+
+        if (
+            not isinstance(self.data, DeferredExecution)
+            and self.flat_args
+            and self.flat_kwargs
+            and self.num_returns == 1
+        ):
+            result, length, width, ip = remote_exec_func.remote(
+                self.func, self.data, *self.args, **self.kwargs
+            )
+            meta = MetaList([length, width, ip])
+            self._set_result(result, meta, 0)
+            return result, meta, 0
+
+        # If there are no subscribers, we still need the result here.
+        self.subscribe()
```

Comment:
I don't really get it, we don't call `.unsubscribe()` at the end of the function, aren't we just adding a dummy reference here that will never be removed?

Also, shouldn't the subscribe call be under this condition to match the comment?
```suggestion
        if self.subscribers == 0:
            # If there are no subscribers, we still need the result here.
            self.subscribe()
```

Reply:
After the execution, this node will never be executed again and this counter has no impact.

Anyway, this new implementation is broken now. Working on it.

Reply:
Should be fixed now. There were an issue after #6883 introduced a new remote function, that returns multiple values.